fv3config/fv3run/_native.py

```diff
@@ -133,20 +138,38 @@ def _temporary_directory(outdir):
         yield outdir
 
 
-@contextlib.contextmanager
-def _log_exceptions(localdir):
-    logger.info("running experiment")
+def _error_context_captured(localdir):
+    out_filename = os.path.join(localdir, STDOUT_FILENAME)
+    err_filename = os.path.join(localdir, STDERR_FILENAME)
+    with open(out_filename, "wb") as out_file, open(err_filename, "wb") as err_file:
+        try:
+            yield out_file, err_file
+        except subprocess.CalledProcessError as e:
+            logger.critical(
+                "Experiment failed. " "Check %s and %s for logs.",
+                STDOUT_FILENAME,
+                STDERR_FILENAME,
+            )
+            raise e
+
+
+def _error_context_uncaptured(localdir):
     try:
-        yield
+        yield sys.stdout, sys.stderr
     except subprocess.CalledProcessError as e:
-        logger.critical(
-            "Experiment failed. " "Check %s and %s for logs.",
-            STDOUT_FILENAME,
-            STDERR_FILENAME,
-        )
+        logger.critical("Experiment failed")
         raise e
 
 
+@contextlib.contextmanager
+def _error_context(localdir: str, capture_output: bool):
```

Comment:
I'd perhaps call this "output context" since it redirects output moreso than it does error handling. It doesn't only redirect stderr.